### PR TITLE
Feat: added quest page

### PR DESCRIPTION
**Pull request by Logan1x**
- added questions in data file
- made reducer
- option color change based on selection

Deploy preview: https://deploy-preview-2--utopia-project.netlify.app/

### CODE REVIEW

File: src/pages/Quest/Quest.jsx [hunk 1]
```diff
@@ -1,8 +1,127 @@
...
+  const { name, questions } = quizData.find((quest) => quest.id === qid);
+
+  const [activeOption, setActiveOption] = useState(0);
+  const [selectedOption, setSelectedOption] = useState("");
+
+  const [index, setIndex] = useState(0);
+  const quest = questions[index];
+
+  const validateAnswer = (quesIndex) => {
+    console.log(
```
Comment (parthpandyappp) [inline src/pages/Quest/Quest.jsx]: Remove this redundant log.

File: src/pages/Quest/Quest.jsx [hunk 2]
```diff
@@ -1,8 +1,127 @@
...
+
+    validateAnswer(quesIndex);
+  };
+
+  const handleOptionClick = (index, option) => {
+    setActiveOption(index);
+    setSelectedOption(option);
+  };
+
+  // console.log(state.score);
```
Comment (parthpandyappp) [inline src/pages/Quest/Quest.jsx:59]: Remove unneccesary comments

File: src/pages/Result/Result.jsx [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+function Result() {
+  const { state, dispatch } = useDataContext();
+  const [error, setError] = useState("");
+
+  if (state.qid !== null) {
+    const { name, questions } = quizData.find(
+      (quest) => quest.id === state.qid
+    );
+  } else if (state.qid === null) {
+    console.log("error");
```
Comment (parthpandyappp) [inline src/pages/Result/Result.jsx]: remove this log

File: src/pages/Result/Result.jsx [hunk 4]
```diff
@@ -0,0 +1,28 @@
...
+    setError("You haven't attempted the quiz yet");
+  }
+
+  return (
+    <div>
+      {error}
```
Comment (parthpandyappp) [inline src/pages/Result/Result.jsx:20]: remove this if not needed.

File: src/pages/Quest/Quest.jsx [hunk 5]
```diff
@@ -1,8 +1,127 @@
...
+
+  const [activeOption, setActiveOption] = useState(0);
+  const [selectedOption, setSelectedOption] = useState("");
+
+  const [index, setIndex] = useState(0);
+  const quest = questions[index];
+
+  const validateAnswer = (quesIndex) => {
+    console.log(
+      `real ans:  ${questions[quesIndex].answer}, your ans:  ${selectedOption}`
```
Comment (manish591) [inline src/pages/Quest/Quest.jsx]: Remove console if not required

**Review by parthpandyappp — APPROVED** [review]
I've left a few comments. Look into it :+1: 
Other looks good to me. Go for merge.

**Review by manish591 — APPROVED** [review]
The code is looking good.

File: src/pages/Result/Result.jsx (hunk 4 above)
Comment (Logan1x) [reply src/pages/Result/Result.jsx:20]: This is been covered in next PR

File: src/pages/Result/Result.jsx (hunk 3 above)
Comment (Logan1x) [reply src/pages/Result/Result.jsx]: Done. Thanks

File: src/pages/Quest/Quest.jsx (hunk 1 above)
Comment (Logan1x) [reply src/pages/Quest/Quest.jsx]: Done. Thanks

File: src/pages/Quest/Quest.jsx (hunk 5 above)
Comment (Logan1x) [reply src/pages/Quest/Quest.jsx]: Done Thanks :)

**Comment by Logan1x** [comment]
Thank You Both, Merging now :)

**Pull request merged**